src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs

```diff
@@ -448,11 +421,10 @@ private static void CacheLoadedAssemblies()
                             continue;
                     }
 
-#if TRACE_ASSEMBLY_LOADING
                     // The "" no-op argument is necessary because WriteLine has multiple overloads, and with two strings
                     // it will be the one that is message/category, rather than format string with
                     System.Diagnostics.Debug.WriteLine("*** Caching classes in {0}", assembly.FullName, "");
```

Comment:
>System.Diagnostics.Debug.WriteLine("*** Caching classes in {0}", assembly.FullName, ""); [](start = 20, length = 88)

The `#if` (which was not, as I understand, typical) has been removed, which means this prior code which was effectively dead will not execute. Honestly I'm not quite sure what the original author had in mind -- the most plausible author to my mind left Microsoft many years ago -- so should not this odd check not go with them, rather than be retained? Unless of course you see some virtue in this message.

Reply:
I left the `Debug.WriteLine` code in because:

1. The `#define` was at the top, meaning this code was compiled in before. I was just removing the hard-coded `#define` and the corresponding `#if`.
2. In release builds, these lines will be compiled out (they even were before, because Debug has `[Conditional("Debug")]` on them. So it doesn't affect the official product.
3. I assumed someone still wanted this logging for debugging purposes.

I can totally remove this code as well, if you want.